### ISSUE TITLE
SDK Uses private Apple APIs

### ISSUE THREAD

**Issue by ivan-ivanovv** [awslabs/aws-sdk-swift#1204]
### Describe the bug

We are using the AWS SDK to access an S3 bucket on our iOS app. Since the latest Xcode update we are getting a "Non-public API usage" error when we try to upload our iOS app to Test Flight: 

```
ITMS-90338: Non-public API usage - The app references non-public symbols in {PROJECT_NAME}: _CCCryptorGCMAddAAD, _CCCryptorGCMAddIV, _CCCryptorGCMFinal, _CCCryptorGCMFinalize, _CCCryptorGCMSetIV. If method names in your source code match the private Apple APIs listed above, altering your method names will help prevent this app from being flagged in future submissions. In addition, note that one or more of the above APIs may be located in a static library that was included with your app. If so, they must be removed. For further information, visit the Technical Support Information at http://developer.apple.com/support/technical/
```

We have identified that after removing the AWS SDK from our app, we stopped getting this warning. 

### Expected Behavior

The AWS SDK shouldn't use any private Apple APIs to apps can be submitted to Test Flight and the App Store. 

### Current Behavior

We are get the following error when we submit our app to Test Flight: 

```
ITMS-90338: Non-public API usage - The app references non-public symbols in {PROJECT_NAME}: _CCCryptorGCMAddAAD, _CCCryptorGCMAddIV, _CCCryptorGCMFinal, _CCCryptorGCMFinalize, _CCCryptorGCMSetIV. If method names in your source code match the private Apple APIs listed above, altering your method names will help prevent this app from being flagged in future submissions. In addition, note that one or more of the above APIs may be located in a static library that was included with your app. If so, they must be removed. For further information, visit the Technical Support Information at http://developer.apple.com/support/technical/
```

Here is a screenshot of the Xcode search for one of the restricted APIs: 
![Screenshot 2023-10-30 at 17 01 56](https://github.com/awslabs/aws-sdk-swift/assets/47433973/332ccb60-f067-4941-a38d-9728ab2a97b3)

### Reproduction Steps

Integrate the SDK into any iOS app and try to access an object from a private S3 bucket (using IAM credentials). 

### AWS SWIFT SDK version used

0.28.0

### Compiler and Version used

Xcode 15.0, swift-driver version: 1.87.1 Apple Swift version 5.9 

### Operating System and version

iOS, any version

**Comment by jbelkins** [comment]
@ivan-ivanovv Thanks for your report.  We will look into this, and we'll update here when we have more information.

**Comment by jbelkins** [comment]
Related issue: https://github.com/awslabs/aws-crt-swift/issues/206

**Comment by jbelkins** [comment]
The amplify-swift SDK also has a related issue: https://github.com/aws-amplify/amplify-swift/issues/3324

Of note, Amplify reports that the problem does not exist in aws-sdk-swift 0.13.0, which requires aws-crt-swift 0.6.1

**Comment by jbelkins** [comment]
@ivan-ivanovv Please try new AWS SDK for Swift version 0.30.0.  It contains a change that should eliminate the message you reported above.

Let us know if this fixes your issue.  Thanks!

**Comment by jbelkins** [comment]
This issue has been confirmed fixed in AWS SDK for Swift 0.30.0.  If you continue to have further problems with this version or later, please file a new issue.
